File: src/multi/first/RedHeadDuck.java
package multi.first;

/**
 * Created by bvvy on 2017/7/26.
 */
public class RedHeadDuck implements Quackable {
    Observable observable;

    public RedHeadDuck() {
        this.observable = new Observable(this);
    }

    @Override
    public void quack() {
        System.out.println("quack");
        notifyObservers();

    }

    @Override
    public void registerObserver(Observer observer) {
        observable.registerObserver(observer);
    }

    @Override
    public void notifyObservers() {
        observable.notifyObservers();
    }
}

File: src/strategy/second/RedHeadDuck.kt
package strategy.second

/**
 * Created by bvvy on 2017/7/13.
 */
open class RedHeadDuck : Duck,Flyable,Quackable {
    override fun fly() {
        print("i am fly ")
    }

    override fun quack() {
        print("gaga")
    }


    override fun swim() {
        print("i m swim")
    }

    override fun display() {
       print("I have a red head")
    }

}